apps/cowswap-frontend/src/modules/usdAmount/apis/getCoingeckoUsdPrice.ts

```diff
@@ -65,26 +56,39 @@ export async function getCoingeckoUsdPrice(currency: Token): Promise<Fraction |
 
   const url = `${BASE_URL}/simple/token_price/${platform}?${new URLSearchParams(params)}`
 
-  return fetchRateLimitted(url)
-    .then((res) => {
-      return res.json()
-    })
+  return fetchRateLimited(url)
+    .then((res) => res.json())
     .catch((error) => {
       if (error.message.includes(FAILED_FETCH_ERROR)) {
-        throw new CoingeckoRateLimitError()
+        throw new RateLimitError()
```

Comment:
you might want to add `cause` :)  https://cowservices.slack.com/archives/C04PLKRBAC8/p1713945604276169